<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProgramMiscEditor</class>
 <widget class="QDialog" name="ProgramMiscEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit miscellaneous program properties</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupFlags">
     <property name="title">
      <string>Flags</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>6</number>
      </property>
      <item>
       <widget class="QCheckBox" name="checkFlagKYONEX">
        <property name="text">
         <string>KYONEX</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkFlagKYONB">
        <property name="text">
         <string>KYONB</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkFlagSSCTL">
        <property name="font">
         <font>
          <underline>true</underline>
         </font>
        </property>
        <property name="toolTip">
         <string>If set, reads random data instead of sample data. Essentially a noise generator.</string>
        </property>
        <property name="text">
         <string>SSCTL</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupEnvelopes">
     <property name="title">
      <string>Envelopes</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>6</number>
      </property>
      <item>
       <widget class="QCheckBox" name="checkEnvLPSLNK">
        <property name="font">
         <font>
          <underline>true</underline>
         </font>
        </property>
        <property name="toolTip">
         <string>&quot;If set, then the envelope transitions to the decay state when the sample loop start address is exceeded.&quot;</string>
        </property>
        <property name="text">
         <string>LPSLNK</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupPitchRegs">
     <property name="title">
      <string>Pitch Registers</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>These seemingly only impact envelopes.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="lblPitchFNS">
          <property name="text">
           <string>FNS</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>spinPitchFNS</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSlider" name="sliderPitchFNS">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>2047</number>
          </property>
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="spinPitchFNS">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>2047</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="lblPitchOCT">
          <property name="font">
           <font>
            <underline>true</underline>
           </font>
          </property>
          <property name="toolTip">
           <string>Apparently, if the sample format is ADPCM and OCT is 2 or higher, OCT is bumped up another octave.</string>
          </property>
          <property name="text">
           <string>Octave</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>spinPitchOCT</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSlider" name="sliderPitchOCT">
          <property name="minimum">
           <number>-8</number>
          </property>
          <property name="maximum">
           <number>7</number>
          </property>
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="spinPitchOCT">
          <property name="minimum">
           <number>-8</number>
          </property>
          <property name="maximum">
           <number>7</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Other</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <widget class="QSlider" name="horizontalSlider">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <underline>true</underline>
           </font>
          </property>
          <property name="toolTip">
           <string>&quot;Not sure what this is representing yet or if it's only 1 byte, but this is present when a program
uses an unusual sample rate as so frequency can be adjusted to finer values than what just
baseNote can represent. Different between versions. See issue #16.&quot;</string>
          </property>
          <property name="text">
           <string>Freq. Adjust</string>
          </property>
          <property name="buddy">
           <cstring>spinBox</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="spinBox"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Unknown</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="2">
         <widget class="QSpinBox" name="spinUnk1">
          <property name="maximum">
           <number>255</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="lblUnk1">
          <property name="text">
           <string>1</string>
          </property>
          <property name="buddy">
           <cstring>spinUnk1</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSlider" name="sliderUnk1">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="lblUnk1Hex">
          <property name="font">
           <font>
            <family>Monospace</family>
           </font>
          </property>
          <property name="text">
           <string>0x0</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>checkFlagKYONEX</tabstop>
  <tabstop>checkFlagKYONB</tabstop>
  <tabstop>checkFlagSSCTL</tabstop>
  <tabstop>checkEnvLPSLNK</tabstop>
  <tabstop>sliderPitchFNS</tabstop>
  <tabstop>spinPitchFNS</tabstop>
  <tabstop>sliderPitchOCT</tabstop>
  <tabstop>spinPitchOCT</tabstop>
  <tabstop>horizontalSlider</tabstop>
  <tabstop>spinBox</tabstop>
  <tabstop>sliderUnk1</tabstop>
  <tabstop>spinUnk1</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ProgramMiscEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>490</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ProgramMiscEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>313</x>
     <y>490</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderUnk1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinUnk1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinUnk1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderUnk1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>296</x>
     <y>382</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderPitchFNS</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinPitchFNS</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>212</y>
    </hint>
    <hint type="destinationlabel">
     <x>368</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinPitchFNS</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderPitchFNS</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>287</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderPitchOCT</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinPitchOCT</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>371</x>
     <y>226</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinPitchOCT</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderPitchOCT</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>358</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>347</x>
     <y>304</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>353</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>281</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
